project/src/main/webapp/emailParse.js

```diff
@@ -151,12 +151,22 @@ class MLModelEmailParse {
         if (index >= 0) {
             const bodyLabel = "Body: ".bold().fontsize(4);
             const highlightedText = `<span class='highlight'>${innerHTML.substring(index, index + answer.length)}</span>`;
-            innerHTML = `${bodyLabel} ${innerHTML.substring(0, index)} ${highlightedText} ${innerHTML.substring(index + answer.length)}`;
+            innerHTML = `${bodyLabel} ${innerHTML.substring(0, index)}${highlightedText}${innerHTML.substring(index + answer.length)}`;
+            let emailWithoutHashCode = this.removeHashCode(innerHTML);
             // Use regular expression to remove empty lines and tabs in body of email.
-            bodyTag.innerHTML = innerHTML.replace(/(^[ \t]*\n)/gm, "");
+            bodyTag.innerHTML = emailWithoutHashCode.replace(/(^[ \t]*\n)/gm, "");
         }
     }
 
+    removeHashCode(innerHTML) {
+        let noHash = innerHTML;
+        const hashedStartIndex = innerHTML.indexOf("--00");
```

Comment:
I recommend using `lastIndexOf` instead just in the very unlikely case there is also "--00" in the email body as well.

http://go/mdn/JavaScript/Reference/Global_Objects/String/lastIndexOf